toolkit/tools/imagegen/configuration/configuration.go

```diff
@@ -93,6 +95,27 @@ func (c *Config) GetDiskContainingPartition(partition *Partition) (disk *Disk) {
 	return nil
 }
 
+// GetKernelCmdLineValue returns the output of a specific option setting in /proc/cmdline
+func GetKernelCmdLineValue(option string, startIndex int) (cmdlineValue string, err error) {
+	const cmdlineFile = "/proc/cmdline"
+
+	content, err := os.ReadFile(cmdlineFile)
+	if err != nil {
+		logger.Log.Errorf("failed to read from %s", cmdlineFile)
+		return
+	}
+
+	cmdlineArgs := strings.Split(string(content), " ")
+	for _, cmdlineArg := range cmdlineArgs {
+		if strings.Contains(cmdlineArg, option) {
+			cmdlineValue = cmdlineArg[startIndex:len(cmdlineArg)]
+			return
```

Comment:
Do we want to handle the case where the same parameter is passed multiple times? (For example you can pass `console=...` multiple times to get multiple consoles).

Maybe just throw an error message if the key shows up more than once?

Reply:
Have added error checking to capture this case.